alf/utils/dist_utils.py

```diff
@@ -1068,7 +1068,8 @@ def get_mode(dist):
     elif isinstance(dist, td.OneHotCategorical) or \
                 isinstance(dist, td.OneHotCategoricalStraightThrough):
         mode = torch.nn.functional.one_hot(
-            torch.argmax(dist.logits, -1), num_classes=dist.logits.shape[-1])
```

Comment:
The original code is consistent with the convention that the ``mode`` has the same dtype as ``samples`` generated from ``dist.sample()``.

Reply:
> The original code is consistent with the convention that the `mode` has the same dtype as `samples` generated from `dist.sample()`.

OK, there is no good way of enforcing this. I'll close this PR.